app/domain/authentication/authn_oidc/update_input_with_username_from_id_token.rb

```diff
@@ -0,0 +1,102 @@
+module Authentication
+  module AuthnOidc
+
+    UpdateInputWithUsernameFromIdToken ||= CommandClass.new(
+      dependencies: {
+        fetch_authenticator_secrets:         Authentication::Util::FetchAuthenticatorSecrets.new,
+        validate_account_exists:             ::Authentication::Security::ValidateAccountExists.new,
+        verify_and_decode_token:             ::Authentication::OAuth::VerifyAndDecodeToken.new,
+        logger:                              Rails.logger
+      },
+      inputs:       %i(authenticator_input)
+    ) do
+
+      extend Forwardable
+      def_delegators :@authenticator_input, :service_id, :authenticator_name,
+                     :account, :username, :webservice, :credentials, :client_ip,
+                     :role
+
+      def call
+        validate_account_exists
+        validate_credentials_include_id_token
+        verify_and_decode_token
+        validate_conjur_username
+        add_username_to_input
+        @authenticator_input
```

Comment:
Change the final two lines to this single line:

```ruby
input_with_username
```
and then change `add_username_to_input` to:

```ruby
      def input_with_username
        @authenticator_input.update(username: conjur_username)
      end
```